R/chromote_session.R

```diff
@@ -561,3 +583,30 @@ ChromoteSession <- R6Class(
     }
   )
 )
+
+
+# Wrapper around ChromoteSession$new() that can return a promise
+create_session <- function(chromote = default_chromote_object(),
+                           width = 992,
+                           height = 1323,
+                           targetId = NULL,
+                           wait_ = TRUE,
+                           auto_events = NULL) {
+
+  session <- ChromoteSession$new(
+    parent = chromote,
+    width = width,
+    height = height,
+    targetId,
+    auto_events = auto_events,
+    wait_ = wait_
+  )
+
+  if (wait_) {
+    session
+  } else {
+    # ChromoteSession$new() must return a ChromoteSession object so we need a
+    # side-channel to return a promise
+    session$init_promise()
```

Comment:
Maybe this should be renamed `get_init_promise()` for clarity? Otherwise it sounds like this is telling it to do the initialization.